src/charm.py

```diff
@@ -183,11 +183,21 @@ def healthy(self) -> bool:
 
         return True
 
-    def _on_update_status(self, _: EventBase) -> None:
+    def _on_update_status(self, event: EventBase) -> None:
         """Handler for `update-status` events."""
         if not self.healthy:
             return
 
+        # NOTE: integration with kafka-broker-rack-awareness charm.
+        # Load current properties set in the charm workload and check
+        # if rack.properties file exists
+        properties = safe_get_file(self.kafka_config.server_properties_filepath)
+        if properties:
+            if self.kafka_config.rack_properties != [] and (
```

Comment:
**nitpick:** To avoid the nesting and falsy empty list, I'd do:

```python
if not properties: 
    return

if not self.kafka_config.rack_properties and (self.kafka_config.server_properties ^ set(properties)):
    self._on_config_changed(event)
```

Reply:
But I don't want to return, the rest of `update_status` should run, even if `properties` is false. I can probably make everything into a single check, but it's going to be a mouthful.